File: packages/typescriptlib/src/test/typescript/serviceConfiguration.test.ts
import {DefaultServiceConfigurationImpl, EMPTY_SERVICE_CONFIGURATION} from "@src/typescript/serviceConfiguration";
import {describe, expect, test} from "vitest";

describe("Service Configuration", () => {
    test("DefaultServiceConfigurationImpl should initialize correctly", () => {
        const properties = new Map<string, any>();
        properties.set("key1", "value1");
        properties.set("key2", "value2");

        const config = new DefaultServiceConfigurationImpl(properties);

        expect(config.propertyMap.size).toBe(2);
        expect(config.propertyMap.get("key1")).toBe("value1");
        expect(config.propertyMap.get("key2")).toBe("value2");
    });

    test("EMPTY_SERVICE_CONFIGURATION should be empty",
         () => { expect(EMPTY_SERVICE_CONFIGURATION.propertyMap.size).toBe(0); });
});

File: packages/pythonlib/src/test/python/service_manager/service_configuration_test.py
"""
    Test thje service_configuration module
"""

from service_configuration import DefaultServiceConfigurationImpl


class TestDefaultServiceConfigurationImpl:
    """
    Test DefaultServiceConfigurationImpl
    """

    def test_constructor_and_poperty_map(self):
        """
        test_constructor and poperty_map method
        """

        # try empty configuration
        service_configuration_to_test = DefaultServiceConfigurationImpl({})
        property_map = service_configuration_to_test.poperty_map()
        assert len(property_map) == 0, "Empty configuration has a 0 length property map"

        # try with set properties
        service_configuration_to_test = DefaultServiceConfigurationImpl(
            {"key1": "value1", "key2": "value2"}
        )
        property_map = service_configuration_to_test.poperty_map()

        assert len(property_map) == 2, "Configuration property map has 2 items"
        assert (
            property_map.get("key1") == "value1"
        ), "Configuration property 1 is correct"
        assert (
            property_map.get("key2") == "value2"
        ), "Configuration property 2 is correct"
